etl/grapher_helpers.py

```diff
@@ -587,3 +587,85 @@ def sanitize_numpy(obj: Any) -> Any:
     elif isinstance(obj, np.ndarray):
         return obj.tolist()
     return obj
+
+
+def add_columns_for_multiindicator_chart(
+    table: catalog.Table,
+    columns_in_chart: List[str],
+    chart_slug: str,
+    suffix_for_titles: Optional[str] = None,
+    columns_to_fill_with_zeros: Optional[List[str]] = None,
+) -> catalog.Table:
+    """Add columns that will be used in a specific multi-indicator (e.g. a stacked area) chart handling issues with
+    missing data.
+
+    There are two common data issues that affect multi-indicator charts:
+    1. Some indicators have missing data that should be zeros. This is a common bad practice done by data producers.
+      For example, in the Statistical Review of World Energy, nuclear power for Australia before 2000 is missing, and
+      Kazakhstan has no data on nuclear energy after 2000. In both cases, those missing values should be zero (since
+      there was no nuclear power for those years in those countries).
+    2. Different indicators have missing data for different years. This makes sense: Some indicators are better informed
+      than others.
+      For example, in the Electricity Mix dataset, Algeria in 2022 has data for renewables but not for fossil fuels.
+      In stacked area charts showing renewable and fossil electricity, Algeria appears as 100% renewable in 2022,
+      which is clearly wrong.
+
+    This function can be used to create columns for all indicators used in a specific multi-indicator chart.
+    It will:
+    * Optionally, fill missing data for certain indicators with zeros.
+      In the example of issue 1, nuclear power could be filled with zeros.
+    * Make nan rows in the new columns if any indicator used in the chart is nan.
+      In the example of issue 2, Algeria's renewable electricity in 2022 would be nan.
+
+    NOTES:
+    * This function assumes a wide-format table.
+    * In the grapher admin, ensure you select "Hide entities with missing data" in the corresponding chart.
+
+    Parameters
+    ----------
+    table : Table
+        Original table.
+    columns_in_chart : List[str]
+        Column names in the relevant multi-indicator chart.
+    chart_slug : str
```

Comment:
Yes, but I think it's also more convenient to see a slug in the variable names (so you know roughly what the chart is about) rather than an ID. If a chart slug changes, nothing will break, the only issue is that the variable names will be a little confusing.
So, both options can cause some confusion, but using grapher IDs seems to cause confusion more often, so I'll keep slugs for now.